src/tests/stress/pedsim/Ped_util/Regions.enc

```diff
@@ -0,0 +1,462 @@
+  (aaa && bbb)
+end
+active class Tiling_box
+  var boxes : [Box]
+  var futures : [Fut[bool]]
+  var futs_left : int
+  def init(agents : [Agent], n : int) : unit
+    match find_extreme(agents) with
+      case (xmax, ymax, xmin, ymin) =>
+        val dx = (xmax - xmin) / n
+        this.boxes = new [Box]( n * n )
+        this.futures = new [Fut[bool]](|this.boxes|)
+        if(n>1) then
+            var x = xmin
+            var y = ymin
+            var i = 0
+            var win = true
+            val dy = (ymax - ymin) / n
+            for xindex <- [0..n - 2] do
+              for yindex <- [0..n - 2] do
+                this.boxes(i) = new Box((x + dx - 1, y + dy - 1), (x, y))
+                if yindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+                end
+                if xindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n))
+                end
+                if xindex != 0 && yindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n + 1))
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+                  ()
+                end
+                i = i + 1
+                y = y + dy
+              end
+              this.boxes(i) = new Box((x + dx - 1, ymax), (x, y))
+              if xindex != 0 then
+                win = win && link((this.boxes)(i), (this.boxes)(i - n))
+                win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+              end
+              win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+              i = i + 1
+              x = x + dx
+              y = ymin
+            end
+            for yindex <- [0..n - 2] do
+              (this.boxes)(i) = new Box((xmax, y + dy - 1), (x, y))
+              if yindex != 0 then
+                win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+                win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+              end
+              win = win && link((this.boxes)(i), (this.boxes)(i - n + 1))
+              win = win && link((this.boxes)(i), (this.boxes)(i - n))
+              i = i + 1
+              y = y + dy
+            end
+            (this.boxes)(i) = new Box((xmax, ymax), (x, y))
+            win = win && link((this.boxes)(i), (this.boxes)(i - n))
+            win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+            win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+        else
+            this.boxes(0) = new Box((xmax,ymax), (xmin,ymin))
+        end
+        for a <- agents do
+          for b <- this.boxes do
+            if get(b!add(a)) then
+                break
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def move(i : int) : unit
+    for id <- [0..|this.boxes| - 1] do
+      (this.futures)(id) = ((this.boxes)(id)) ! move()
+    end
+    for f <- this.futures do
+      get(f)
```

Comment:
No

First of all, you should know that all the boxes can work on fulfilling their futures in parallel, which means that the execution is not serialised. In this active object the execution is serialised, but that's the only reason active objects are thread safe at all.

I have explained this design in my report, and in several of our meetings earlier, as well as shown that this design indeed is capable parallel execution.

Secondly, it's also a requirement of the problem specification that all active objects synchronise between time steps, and they do this by fulfilling futures.